nima/webclient/webclient/src/main/java/io/helidon/nima/webclient/WebClient.java

```diff
@@ -67,12 +70,14 @@ static <T> T builder(Protocol<T> protocol) {
 
         private URI baseUri;
         private Tls tls;
-        private SocketOptions channelOptions;
+        private final SocketOptions.Builder socketOptionsBuilder = SocketOptions.builder();
         private DnsResolver dnsResolver;
         private DnsAddressLookup dnsAddressLookup;
         private boolean followRedirect;
         private int maxRedirect;
         private WritableHeaders<?> defaultHeaders = WritableHeaders.create();
+        private Duration connectTimeout;
```

Comment:
We should not need to store these as fields, we can simply update the `SocketOptions.Builder`